Python/Product/PythonTools/PythonTools/Intellisense/ProjectAnalyzer.cs

```diff
@@ -2264,9 +2341,19 @@ class ExprWalker : PythonWalker {
                 fileId = analysis.Entry.FileId,
             };
 
-            var resp = await analysis.Entry.Analyzer.SendRequestAsync(req, timeout: timeout).ConfigureAwait(false);
-
-            return resp?.expression;
+            return analysis.Entry.Analyzer.EnsureSingleRequest(
+                typeof(AP.ExpressionForDataTipRequest),
+                req.expr,
+                t => t == req.expr,
```

Comment:
Shouldn't it also compare the location? What if it's 2 different request for tooltips for the same expression, but in different locations which would produce different results?